pkg/reconciler/shared/tektonconfig/tektonconfig.go

```diff
@@ -111,6 +111,15 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tc *v1alpha1.TektonConfi
 		return nil
 	}
 
+	// run pre upgrade
+	isUpgradePerformed, err := r.upgrade.RunPreUpgrade(ctx)
+	if err != nil {
+		return err
+	}
+	if isUpgradePerformed {
```

Comment:
is not this should be `!`

Reply:
If upgrade performed, TektonConfig CR status get updated
and in `RECONCILE_AGAIN_ERR`
```
// RECONCILE_AGAIN_ERR
// When we updates spec or status we reconcile again and then proceed so
// that we proceed ahead with updated object
RECONCILE_AGAIN_ERR = fmt.Errorf("reconcile again and proceed")
```

So I am calling reconcile again to get the most recent TektonConfig CR, if upgrade executed

Reply:
ohk cool